### PR TITLE
Updating roadmap to reflect current status.

### PR DESCRIPTION
**Pull request by nickerso**
Addresses #330.

### CODE REVIEW

File: docs/roadmap.rst [hunk 1]
```diff
@@ -61,110 +52,64 @@ Requirements
...
-   #. Using Buildbot for CI framework
-   #. Interact with GitHub to process changes
-   #. Make Buildbot status available from autotest.bioeng.auckland.ac.nz/libcellml-buildbot/
-
-#. Document the development process/workflow.
-
-   #. Code contribution document
-   #. Developer setup document
-   #. Code review document
+Milestone 1: python bindings, validation, code generation, and documentation
```
Comment (agarny) [inline docs/roadmap.rst]: `Python` rather than `python`.

File: docs/roadmap.rst [hunk 2]
```diff
@@ -61,110 +52,64 @@ Requirements
...
-#. Language support: C++, Python, Java, MATLAB.
 #. Documentation available.
 
    #. API.
    #. Tutorials/documented code examples.
-   #. Integrating libCellML into various common IDEs (Visual Studio, Eclipse, Qt Creator, NetBeans...).
-
-Milestone 2: toward simulation support (timeframe: ? months)
-============================================================
+   #. Integrating libCellML into common IDEs (Visual Studio, Qt Creator, PyCharm).
```
Comment (agarny) [inline docs/roadmap.rst]: `..., and PyCharm` rather than `..., PyCharm`.

File: docs/roadmap.rst [hunk 3]
```diff
@@ -61,110 +52,64 @@ Requirements
...
-#. Improved support for model authoring/editing/manipulation.
+   #. Checking units within mathematical expressions.
+   #. Debugging assistance for model authors regarding units.
 
-   #. Provide an events system to monitor changes in the model.
-   #. Provide access to data contained in CellML models in external namespaces (RDF, extensions, etc.).
+#. JavaScript.
 
-#. Documentation, documentation and more documentation!
+   #. Use emscripten to create a JavaScript API for libCellML.
```
Comment (agarny) [inline docs/roadmap.rst]: `Emscripten` rather than `emscripten`.

File: docs/roadmap.rst [hunk 4]
```diff
@@ -61,110 +52,64 @@ Requirements
...
+   #. Provide a suitable packaged version of the JavaScript API for integration in common JavaScript environments (node, webpack).
```
Comment (agarny) [inline docs/roadmap.rst]: `... (e.g., Node, Webpack).`

File: docs/roadmap.rst [hunk 5]
```diff
@@ -61,110 +52,64 @@ Requirements
...
+   #. Document the installation and use of the JavaScript API.
 
-Milestone 3: functional library for tool developers (timeframe: ? months)
-=========================================================================
+#. resets.
```
Comment (agarny) [inline docs/roadmap.rst]: `Resets` rather than `resets`.

**Comment by agarny** [comment]
Looks good pending some minor corrections.

**Comment by agarny** [comment]
Still looks good to me. :)

**Comment by nickerso** [comment]
Closes #273 and #330.

**Pull request merged**